src/Acuminator/Acuminator.Analyzers/StaticAnalysis/LegacyBqlConstant/LegacyBqlConstantFix.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using Acuminator.Analyzers.StaticAnalysis.LegacyBqlField;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Acuminator.Analyzers.StaticAnalysis.LegacyBqlConstant
+{
+	[ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+	public class LegacyBqlConstantFix : CodeFixProvider
+	{
+		public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+			ImmutableArray.Create(Descriptors.PX1061_LegacyBqlConstant.Id);
+
+		public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+		public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+			var classNode = root.FindNode(context.Span).FirstAncestorOrSelf<ClassDeclarationSyntax>();
+			if (classNode == null)
+				return;
+
+			var diagnostic = context.Diagnostics.FirstOrDefault(d => d.Id == Descriptors.PX1061_LegacyBqlConstant.Id);
+			if (diagnostic == null)
+				return;
+
+			if (diagnostic.Properties == null
+				|| !diagnostic.Properties.TryGetValue(LegacyBqlConstantAnalyzer.CorrespondingType, out string typeName)
+				|| String.IsNullOrEmpty(typeName))
+			{
+				return;
+			}
+
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			string title = nameof(Resources.PX1061Fix).GetLocalized().ToString();
+
+			context.RegisterCodeFix(
+				CodeAction.Create(
+					title,
+					c => Task.FromResult(
```

Comment:
It is better to make it async.

Reply:
so how should I change it exactly? I peeked this approach in some other diagnostic and copied it as-is

Reply:
I would suggest to use it inside Task.Run.

Reply:
done

Reply:
`Task.Run` in this case is not recommended, as stated in our Best Practices.
You can find an example here: https://msdn.microsoft.com/en-us/magazine/dn904670.aspx?f=255&MSPPError=-2147217396

Reply:
@dropsonic then what should be used for code fix which doesn't contain any `await` operations?
Is there a difference between returning `Task.Run` and `await/async`? Maybe the synchronious pattern is ok?

Reply:
Ok, I studied this question a little bit further, and it seems that there is no straight answer.
If there is no async code inside the code action itself, we don't know what's best.
`Task.Run` can lead to thread pool starvation, but it heavily depends on the task scheduler that VS uses.
On the other side, `Task.FromResult` can lead to blocking the UI thread since this code action can be executed for multiple pieces of code.

Reply:
I'm not sure if `Task.FromResult` it will lead to hanging of the UI. I have a suspicion that the code fix is already executed asynchrounously. At least the code in BatchFixAllProvider can lead to such conclusion. So I think in the end the `Tas.Run` is not needed.

Reply:
so should I return ```Task.FromResult``` back?

Reply:
@Derrekam seems so. Sorry for that.

Reply:
@Derrekam Yes, seems that the `FromResult` is alright

Reply:
done